buildbot/master/files/config/factories.py

```diff
@@ -109,6 +113,17 @@ def make_step(self, command):
             elif arg == './etc/ci/upload_nightly.sh':
                 step_kwargs['logEnviron'] = False
                 step_env += envs.upload_nightly
+                if self.is_windows:
+                    # s3cmd on Windows only works within msys
+                    step_env['MSYSTEM'] = 'MSYS'
```

Comment:
hrm, when I was using `envs.Environment`, the debugging seemed to indicate that it wasn't overwriting it - we were still getting the old value in testing. I can give it another try, though, and see what happens.